io-engine/src/core/env.rs

```diff
@@ -292,11 +293,12 @@ impl MayastorFeatures {
         let lvm = env::var("ENABLE_LVM").as_deref() == Ok("true");
         let snapshot_rebuild =
             env::var("ENABLE_SNAPSHOT_REBUILD").as_deref() == Ok("true");
-
+        let rdma_capable_target = is_mayastor_nvmf_tgt_rdma_capable();
```

Comment:
hmm, we're conflating things a bit here, features exposes if it's capable, but here we're checking if it's capable and the target is listening?
I think we should separate those two things

Reply:
The definition of "being rdma capable" for a running mayastor instance target here is whether it has rdma transport created and can take rdma connections, else it's a usual tcp mayastor target instance . What can we separate here?

Reply:
or should we simply expose this as being capable in meaning that the target can take `--enable-rdma` flag?

Reply:
A nuance here is that `ENABLE_RDMA` env could be set but it doesn't guarantee that the target will eventually be rdma capable, until we can successfully setup the target for rdma in runtime.

Reply:
I think we can simply expose whether we're rdma capable for now, which would be the "--enable-rdma" feature gate.
If we want to know if we're listening for the transport RDMA then we can add more messages to expose runtime information.
Example, we could expose subsystem listeners.

Reply:
That might make control plane's life difficult as it would see the io-engine is rdma capable upon registration, but rdma configuration could've actually failed.
I don't see any issue in exposing capability based on complete (one-time) runtime info, considering this is a little different due to "enablement" isn't guaranteeing "usability".

Reply:
Let's image we start binding to more than one network (which I'm sure we will at some point), how would you then expose your rdma feature if you succeeded listening on only 1 of them?
What if we want start listening on networks dynamically?
IMHO, conflating capability with usability is what'd make control plane's life difficult.

Reply:
On the other hand, we'd be complicating API if we want to get real view of capability.

>What if we want start listening on networks dynamically?

What does this mean? I imagine it would certainly come with much bigger challenges compared to exposing capability.

Reply:
I'm not sure how adding another message containing the listener and the listen transport complicates it?

Reply:
That new message would most likely be fetched with a new gRPC call to io-engine if control plane wants to do something, which basically means the capability info we get on registration info isn't reliable already.

Reply:
so that's essentially doing the same thing, just splitting the info into two messages and stashing in same grpc(registration). (¬_¬)

Reply:
let's go ahead with simple env based capability exposure.

Reply:
But then it won't be sufficient for how you intended to use it?
If you're really adverse to split it into two flags for some reason, then keep it as you had it (reading rdma listener) but please rename the proto to something that makes it very clear what's it doing so we don't make mistake of using it incorrectly in the future.
Ex: nvmfRdmaActive

Reply:
> But then it won't be sufficient for how you intended to use it?

I don't see any immediate use-case in control-plane where we might have to do something based on concrete evidence of active rdma in io-engine. Control-plane logic as it stands today is driven mainly by uri scheme which inherently indicates active rdma. So I can rename it to `rdmaCapableIoEngine` or let it be `rdmaCapableTarget`

Reply:
If there's no need for active rdma port check, then rdmaCapableIoEngine or even just rdma sounds good.

Reply:
changed to rdmaCapableIoEngine.